plugin/utils/gapic_utils.py

```diff
@@ -134,15 +144,15 @@ def reconstruct_gapic_yaml(gapic_config, request):  # noqa: C901
     gapic_config['config_schema_version'] += '-reconstructed'
 
     # Sort existing collections in a dictionary so we can look up by
-    # entity names. (This makes it easier to avoid plowing over stuff that
+    # name patterns. (This makes it easier to avoid plowing over stuff that
     # is explicitly defined in YAML.)
-    collections = OrderedDict([(i['entity_name'], i) for i in
+    collections = OrderedDict([(i['name_pattern'], i) for i in
```

Comment:
In some APIs, the GAPIC config's `entity_name`, e.g. `Topic`, differs from what we would expect the entity_name should be. The `name_pattern` is the only thing that we can expect to be the same between the GAPIC config and the proto file.